### PR TITLE
Add wrappable token

### PR DESCRIPTION
**Pull request by willjgriff**
Add wrappable token allowing the TokenManager to mint/burn the MiniMeToken when the wrappable token is deposited.

Includes tests.

### CODE REVIEW

File: contracts/HookedTokenManager.sol [hunk 1]
```diff
@@ -240,6 +245,30 @@ contract HookedTokenManager is ITokenController, IForwarder, AragonApp {
...
+    /**
+    * @notice Unwrap @tokenAmount(self.token(): address, _amount, false) to receive @tokenAmount(self.wrappableToken(): address, _amount, false)
+    * @param _amount Amount of tokens to unwrap
+    */
+    function unwrap(uint256 _amount) public {
+        require(wrappableToken != address(0), ERROR_NO_WRAPPABLE_TOKEN);
+        require(msg.sender != address(this), ERROR_MINT_RECEIVER_IS_TM);
+
+        wrappableToken.safeTransfer(msg.sender, _amount);
+        _burn(msg.sender, _amount);
```
Comment (0xGabi) [inline contracts/HookedTokenManager.sol:269]: Wonder if it's a better security practice to call the `_burn` method before we do the token transfer here?

**Review by 0xGabi — APPROVED** [review]
LGTM! 🙌

Left one suggestion.

File: contracts/HookedTokenManager.sol [hunk 2]
```diff
@@ -240,6 +245,30 @@ contract HookedTokenManager is ITokenController, IForwarder, AragonApp {
         emit RevokeVesting(_holder, _vestingId, nonVested);
     }
 
+    /**
+    * @notice Wrap @tokenAmount(self.wrappableToken(): address, _amount, false) to receive @tokenAmount(self.token(): address, _amount, false)
+    * @param _amount Amount of tokens to wrap
+    */
+    function wrap(uint256 _amount) public {
```
Comment (fabriziovigevani) [inline contracts/HookedTokenManager.sol:252]: Should we make them `external`?

**Review by fabriziovigevani — COMMENTED** [review]
Looks good!

Wondering if we should require that `wrappableToken` is not set when `issuing` tokens so that we ensure a 1 to 1 correspondence.

**Comment by willjgriff** [comment]
> Wondering if we should require that `wrappableToken` is not set when `issuing` tokens so that we ensure a 1 to 1 correspondence.

I did think about this but I figured let the DAO have the option if it wants. Don't know why it would choose to do that but it's the token depositors that will have the ability to do it, maybe there's a reason we can't think of. They could always upgrade the contract to add it anyway so figure okay to leave it.

**Comment by fabriziovigevani** [comment]
Yeah that's also something i was thinking of, but wasn't sure if we would like to restrict that in this case. I think it's okay if we give the option to the DAO as well.

**Pull request merged**